test/e2e/external_gateways.go

```diff
@@ -1008,17 +1134,57 @@ EOF
 				framework.ExpectNoError(err, "failed to setup FRR peer %s in %s", a, containerName)
 			}
 		}
-		cmd := []string{"docker", "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
+		cmd := []string{containerRuntime, "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
 		_, err := runCommand(cmd...)
 		framework.ExpectNoError(err, "failed to start frr in %s", containerName)
 	}
 }
 
 func isBFDPaired(container, peer string) bool {
-	res, err := runCommand("docker", "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
+	res, err := runCommand(containerRuntime, "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
 	framework.ExpectNoError(err, "failed to check bfd status in %s", container)
 	if strings.Contains(res, "Status: up") {
 		return true
 	}
 	return false
 }
+
+// When running on host network we clean the routes and ips we added previously
+func cleanRoutesAndIPs(containerName string, addressesv4, addressesv6 gatewayTestIPs) {
+	testIPv6 := false
+	testIPv4 := false
+
+	if addressesv6.srcPodIP != "" && addressesv6.nodeIP != "" {
+		testIPv6 = true
+	}
+	if addressesv4.srcPodIP != "" && addressesv4.nodeIP != "" {
+		testIPv4 = true
+	}
+
+	if !testIPv4 && !testIPv6 {
+		return
```

Comment:
Wouldn't that be a failure? (I know it's not this function's task to deal with that, but if we aren't testing either of the 2 address families, what are we testing, then?)

Reply:
being part of a broader cleanup this shouldn't fail here to let the AfterEach continue